linode/resource_linode_instance.go

```diff
@@ -1160,16 +1160,44 @@ func resourceLinodeInstanceUpdate(d *schema.ResourceData, meta interface{}) erro
 		d.Partial(false)
 	}
 
-	if d.HasChange("type") {
-		if err = changeInstanceType(&client, instance, d.Get("type").(string), d); err != nil {
-			return err
-		}
-		d.Set("type", d.Get("type").(string))
-	}
+	var rebootInstance bool
+	var diskIDLabelMap map[string]int
 
 	tfDisksOld, tfDisksNew := d.GetChange("disk")
 
-	rebootInstance, diskIDLabelMap, err := updateInstanceDisks(client, d, *instance, tfDisksOld, tfDisksNew)
+	tfDisksOldInterface := tfDisksOld.([]interface{})
```

Comment:
This adds complexity to a function that is already very large; is it possible to put this into its own function?

Reply:
Yes, 💯 . I'm adding getDiskSizeChange to linode_instance_helpers to deal with this.